examples/charts/rendering-spec/src/index.js

```diff
@@ -0,0 +1,124 @@
+import ChartsEmbedSDK from "@mongodb-js/charts-embed-dom";
+import "@alenaksu/json-viewer";
+
+const sdk = new ChartsEmbedSDK({
+  baseUrl: "https://charts.mongodb.com/charts-embedding-examples-wgffp", // Optional: ~REPLACE~ with the Base URL from your Embed Chart dialog
+});
+
+const chart = sdk.createChart({
+  chartId: "735cfa75-15b8-483a-bc2e-7c6659511c7c", // Optional: ~REPLACE~ with the Chart ID from your Embed Chart dialog
+  height: "700px",
+});
+
+async function renderChart() {
+  await chart.render(document.getElementById("chart"));
+
+  /*
+    Collecting all the data from the form and based on it preparing the rendering spec
+    that will be sent to chart.setRenderingSpecOverride() to apply customizations on the embedded chart
+  */
+  document.forms["form"].addEventListener("submit", async function (e) {
+    e.preventDefault();
+
+    // get all the values from the form
+    const formMarkup = new FormData(e.target);
+    const formData = Object.fromEntries(formMarkup);
+
+    const { title, description, labelX, labelY, labelSize, colorPalette } =
+      formData;
+
+    // prepare the rendering spec document
+    const renderingSpec = {
+      version: 1,
+      ...(title ? { title } : {}),
+      ...(description ? { description } : {}),
+    };
+
+    if (labelX || labelY) {
+      renderingSpec.channels = {
+        ...(labelX ? { x: { labelOverride: labelX } } : {}),
+        ...(labelY ? { y: { labelOverride: labelY } } : {}),
+      };
```

Comment:
I think I was away, but this won't work for candlestick charts because the x and y channels are named differently right?

If so, we should somehow communicate that.

Reply:
It will work for Candlestick for the field on the `x` channel. For the `y` axis, it should be an axis labelOverride for `y` axis, as there are multiple fields/channels on the axis. This example will work for all chart types that have `x` and `y` channels. Users should check the renderingSpec docs for specific configurations of embedded charts.

There is a note in the example readme `Customization options are specific to chart types so if you are embedding a different chart type, make sure to check our rendering spec documentation`. Plus, there are comments in the `getCustomizableAxes()` and `getChannels()` code base snippets.

I can add a comment to this exact place in the code base, that the renderingSpec is specific for chart type and chart configuration? Do you think this will be enough to make things more clear? I'm open to suggestions as well. Thanks for the feedback.